pkg/generate/git/repository.go

```diff
@@ -337,6 +348,42 @@ func command(w io.Writer, name, dir string, env []string, args ...string) (stdou
 	return
 }
 
+// timedCommand executes an external command in the given directory with a timeout.
+// The command's standard out and error are trimmed and returned as strings
+// It may return the type *GitError if the command itself fails.
+func timedCommand(timeout time.Duration, w io.Writer, name, dir string, env []string, args ...string) (stdout, stderr string, err error) {
```

Comment:
Code would be shorter overall if you made `Command` call this.  move the entire `command` method in here and used 

```go
cmd.Start()

if timeout > 0{
   select{
   case <-cmd.Wait():
   case <- time.After(timeout):
   }
} else{
   select{
   case <- cmd.Wait():
   }
}
```


right?